src/logcollector/logcollector.c

```diff
@@ -1454,8 +1464,29 @@ static IT_control remove_duplicates(logreader *current, int i, int j) {
                 }
             }
 
-            if (current != dup && dup->file && !strcmp(current->file, dup->file)) {
-                mwarn(DUP_FILE, current->file);
+            struct stat statCurrent, statDup;
+
+            if (stat(current->file, &statCurrent) < 0){
+                merror("Couldn't stat file '%s'", current->file);
+                d_control = LEAVE_IT;
+                break;
+            }
+
+            if (stat(dup->file, &statDup) < 0){
```

Comment:
We are not controlling that `dup->file` can point to a command instead of a file. In that case, with the following configuration:

```xml
<localfile>
    <log_format>command</log_format>
    <command>df -P</command>
    <frequency>360</frequency>
  </localfile>
```

We get this error at startup:
```
2019/08/27 08:37:45 ossec-logcollector: ERROR: Couldn’t stat file ‘df -P’
```

Reply:
Fixed https://github.com/wazuh/wazuh/pull/3867/commits/f2cd15d0144c834a672d9570b771952510f53026